### DISCUSSION TITLE
Is there any list of 'high-level abstraction' libraries using undici?

### DISCUSSION THREAD

**Discussion by r37r0m0d3l** [nodejs/undici#1195] [Q&A]
The question is in title. There is fetch standard and many request-response libraries like: axios, request, superagent, got, ky etc. I look at dependency repos and could not find any such library with request cancellation, retry functionality, progress events. Any suggestions?

**Comment by mcollina** [comment]
This library has:

1. cancelation support
2. retry logic

It lacks progress events.

**Reply by r37r0m0d3l** [reply]
Thank you for your answer. I meant all possible options. In my company, we chose the 'under-the-hood' library for SDK. And want something that we won't change in a few months. 'undici' looks very promising but lacks some features and is very 'low-level' (as fetch) which results in many copy-paste code or reinventing the wheel. I think that's how it was intended. Also, you can add 'undici' to this comparison table if you think it's a good place for it [https://github.com/sindresorhus/got#comparison](https://github.com/sindresorhus/got#comparison).
